### ISSUE TITLE
stage1 training issue

### ISSUE THREAD

**Issue by zhiqiuiyiye** [yl4579/StyleTTS2#175]
Hi, I'm training the styletts2 on a new language Thai, when I trained epoch 7 , I found the loss were been Nan, and the g_loss seems increasing when traing. I want to know what will cause this problem. Here is my log, training loss.
![微信截图_20231228094528](https://github.com/yl4579/StyleTTS2/assets/151741473/968ebea0-7a16-4419-8f1b-8bf8bf4c22ce)
![微信截图_20231228094433](https://github.com/yl4579/StyleTTS2/assets/151741473/e5e51a0d-b803-4523-90d0-68efdd0b04f7)

**Comment by yl4579** [comment]
Sorry for the late reply. I was quite busy recently. Have you checked #10 and #11? Did you use mixed precision as well?

**Comment by zhiqiuiyiye** [comment]
thanks for your reply, I have fixed this issue, may caused too small batch size

**Comment by akshatgarg99** [comment]
Were you able to do it? I was trying to train but was facing some issue. Can we discuss?

**Comment by RillmentGames** [comment]
No that didn't work :( the loss made some strange moves and eventually ended with Nan.
<img width="1042" alt="Stts2NanProblem" src="https://github.com/yl4579/StyleTTS2/assets/66327794/eda83ed8-8623-4c95-a776-c821e02c82ac">

**Comment by RillmentGames** [comment]
Integrating [PhaseAug](https://github.com/maum-ai/phaseaug) and using batch_percentage=1.0 with Batch=2, fixed it for me. 
PhaseAug tries to address the overfitting issue by randomly rotating the phase of each frequency bin. 
The gen error still creeps up but very slowly now and audio quality becomes quite nice after 2 epochs:
```
...
    aug = PhaseAug()
    gl = GeneratorLoss(model.mpd, model.msd, aug).to(device)
    dl = DiscriminatorLoss(model.mpd, model.msd, aug).to(device)
...
class GeneratorLoss(torch.nn.Module):

    def __init__(self, mpd, msd, aug):
        super(GeneratorLoss, self).__init__()
        self.mpd = mpd
        self.msd = msd
        self.aug = aug  
        
    def forward(self, y, y_hat):
        y, y_hat = self.aug.forward_sync(y, y_hat)                 #               <--- Augment here
        y_df_hat_r, y_df_hat_g, fmap_f_r, fmap_f_g = self.mpd(y, y_hat)
        y_ds_hat_r, y_ds_hat_g, fmap_s_r, fmap_s_g = self.msd(y, y_hat)
        loss_fm_f = feature_loss(fmap_f_r, fmap_f_g)
        loss_fm_s = feature_loss(fmap_s_r, fmap_s_g)
        loss_gen_f, losses_gen_f = generator_loss(y_df_hat_g)
        loss_gen_s, losses_gen_s = generator_loss(y_ds_hat_g)

        loss_rel = generator_TPRLS_loss(y_df_hat_r, y_df_hat_g) + generator_TPRLS_loss(y_ds_hat_r, y_ds_hat_g)
        
        loss_gen_all = loss_gen_s + loss_gen_f + 1.0*loss_fm_s + 1.0*loss_fm_f + loss_rel
        
        return loss_gen_all.mean()
    
class DiscriminatorLoss(torch.nn.Module):

    def __init__(self, mpd, msd, aug):
        super(DiscriminatorLoss, self).__init__()
        self.aug = aug
        self.mpd = mpd
        self.msd = msd
        
    def forward(self, y, y_hat):
        y, y_hat = self.aug.forward_sync(y, y_hat.detach())    #                   <--- Augment here
        # MPD
        y_df_hat_r, y_df_hat_g, _, _ = self.mpd(y, y_hat)
        loss_disc_f, losses_disc_f_r, losses_disc_f_g = discriminator_loss(y_df_hat_r, y_df_hat_g)
        # MSD
        y_ds_hat_r, y_ds_hat_g, _, _ = self.msd(y, y_hat)
        loss_disc_s, losses_disc_s_r, losses_disc_s_g = discriminator_loss(y_ds_hat_r, y_ds_hat_g)
        
        loss_rel = discriminator_TPRLS_loss(y_df_hat_r, y_df_hat_g) + discriminator_TPRLS_loss(y_ds_hat_r, y_ds_hat_g)


        d_loss = loss_disc_s + loss_disc_f + loss_rel
        
        return d_loss.mean()
```
